src/hydration/thirst.ts

```diff
@@ -1,18 +1,97 @@
 import { Bukkit, GameMode } from 'org.bukkit';
 import { Player } from 'org.bukkit.entity';
 import { PotionEffect, PotionEffectType } from 'org.bukkit.potion';
+import { addHydration, getHydration } from './expbar';
 
-// How often players' hydration level is decreased (seconds)
-const THIRST_CYCLE_DURATION = 60;
+/**
+ * How often players' hydration level is decreased (seconds)
+ * If duration is 60s and hydration -0.01, the bar will last 100 mins
+ */
+const THIRST_CYCLE_DURATION = 3;
 const THIRST_CYCLE_HYDRATION = -0.01;
-// If duration is 60s and hydration -0.1, the bar will last 100 mins
 
-// How often dehydration effects are aplied (seconds)
+/**
+ * How often dehydration effects are aplied (seconds)
+ */
 const DEHYDRATION_CYCLE_DURATION = 10;
 
-const DEHYDRATION_LEVEL = 0.2;
-const MODERATE_DEHYDRATION_LEVEL = 0.09;
-const SEVERE_DEHYDRATION_LEVEL = 0.06;
+const DehydrationLevel = {
+  MILD: 0.2,
+  MODERATE: 0.09,
+  SEVERE: 0.06,
+};
+
+const dehydratedPlayers = new Set<Player>();
+
+// Reduce the hydration of the players
+setInterval(() => {
+  for (const player of Bukkit.server.onlinePlayers) {
+    reduceHydrationOf(player);
+  }
+}, THIRST_CYCLE_DURATION * 1000);
+
+function reduceHydrationOf(player: Player) {
+  if (player.gameMode !== GameMode.SURVIVAL) return;
+
+  addHydration(player, THIRST_CYCLE_HYDRATION);
+  const hydration = getHydration(player);
+
+  if (hydration < DehydrationLevel.MILD) {
+    becomeDehydrated(player, hydration);
+  }
+}
+
+function becomeDehydrated(player: Player, hydration: number) {
+  dehydratedPlayers.add(player);
+
+  const msg = getDehydrationMessage(hydration);
+  player.sendTitle('', msg, 10, 40, 10);
+}
+
+function getDehydrationMessage(hydration: number) {
```

Comment:
Shouldn't this be made with the translation system?

Reply:
This was made before translation system, so translations can be added later